### ISSUE TITLE
`<Sidebar>` component's `<nav>` should have an accessible name

### ISSUE THREAD

**Issue by isner** [dequelabs/cauldron#327]
### The Problem

The `<Sidebar>` component's props allow users to set arbitrary `...other` attributes on the `<ul>` element. However, users have no way to set attributes on the `<nav>`. An `aria-label` attribute on the `<nav>` element is important to distinguish the sidebar's navigation landmark region from other navigation landmark regions that might be elsewhere on the page.

https://github.com/dequelabs/cauldron/blob/f4b6cdd082b4f2f0d692d8e535951486988acedb/packages/react/src/components/SideBar/SideBar.tsx#L10-L15

https://github.com/dequelabs/cauldron/blob/f4b6cdd082b4f2f0d692d8e535951486988acedb/packages/react/src/components/SideBar/SideBar.tsx#L126-L137

### Possible Solution

Have the `<Sidebar>` component accept an optional `label` prop whose value is used to define an `aria-label` attribute on the component's `<nav>` element.

**Comment by schne324** [comment]
We may want to not limit it to `aria-label` and instead support a `navProps` prop. This is a common enough pattern that I think it makes sense. For example, material-ui has an `inputProps` prop on their [`<TextField />` component](https://mui.com/api/text-field/) which allows the caller to pass arbitrary props through to the actual input element rendered by `<TextField />`
